helix-term/src/ui/overlay.rs

```diff
@@ -15,6 +15,12 @@ pub struct Overlay<T> {
     pub calc_child_size: Box<dyn Fn(Rect) -> Rect>,
 }
 
+// TODO: For this to be sound, all of the various functions
+// have to now be marked as send + sync + 'static. Annoying,
+// and something I'll look into with steel.
+unsafe impl<T> Send for Overlay<T> {}
+unsafe impl<T> Sync for Overlay<T> {}
```

Comment:
Can we centralize all of these definitions in one file and add a config attr so they're only compiled with steel?